doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
 
 For patch releases we never allow changes in the public API, unless the change is specifically to fix a bug that aligns the actual behavior of the code with the publicly documented behavior.
 
+Only release branches (more on those later) include the full major.minor.patch
+version numbers included with releases. The major and minor version of the _master_ and
+_develop_ branches do not include a patch number and always have a SNAPSHOT tag.
+The _master_ and _develop_ versions always matches the pattern *MAJOR*.*MINOR*-SNAPSHOT.
```

Comment:
This allows it to be easily pulled into the _develop_ branch (and reduce the amount of versioning commits in the history too).